packages/interface-transport/src/index.ts

```diff
@@ -78,16 +79,21 @@ export interface UpgraderEvents {
   'connectionEnd': CustomEvent<Connection>
 }
 
+export interface UpgraderOptions {
+  skipEncryption?: boolean
+  muxerFactory?: StreamMuxerFactory | undefined
```

Comment:
Skipping connection protection should be made explicit.

It's true the transports we have right now that have their own encrypters and muxers also don't support connection protection but that may not always be the case.

We also then won't end up deciding whether or not to protect a connection based on the presence of a muxer factory in the options as we currently do in https://github.com/libp2p/js-libp2p/pull/1395/files#diff-c199200bb38fc06cbe1654f84bb2cfaed1cf0aa9bc9ed71f646932604cbd9252R266-R272

```suggestion
  skipEncryption?: boolean
  skipProtection?: boolean
  muxerFactory?: StreamMuxerFactory | undefined
```